src/modules/ethereum/transactions/gasPrices/optimistic/fee.ts

```diff
@@ -69,7 +92,20 @@ export async function createOptimisticFee({
 
   const l1_fee_est = Math.round(l1 * l1GasEstimation);
   return {
-    maxFee: l2 * Number(gas) + Math.round(l1_fee_est * 1.25), // Optimism guarantees that there should not be one-time jumps of l1 component of gas estimation more than 25%
-    estimatedFee: l2 * Number(gas) + l1_fee_est,
+    maxFee:
+      (eip1559GasPrice
+        ? Number(eip1559GasPrice.max_fee)
+        : Number(classicGasPrice)) *
+        Number(gas) +
+      Math.round(l1_fee_est * 1.25), // Optimism guarantees that there should not be one-time jumps of l1 component of gas estimation more than 25%
+    estimatedFee:
+      (eip1559GasPrice
+        ? Math.min(
+            Number(eip1559GasPrice.max_fee),
+            Number(base_fee) + Number(eip1559GasPrice.priority_fee)
+          )
+        : Number(classicGasPrice)) *
+        Number(gas) +
+      l1_fee_est,
   };
```

Comment:
I think this might be more readable:
```ts
  const maxFee = eip1559GasPrice
    ? Number(eip1559GasPrice.max_fee)
    : Number(classicGasPrice);
  const expectedFee = eip1559GasPrice
    ? Math.min(
        Number(eip1559GasPrice.max_fee),
        Number(base_fee) + Number(eip1559GasPrice.priority_fee)
      )
    : Number(classicGasPrice);
  return {
    maxFee: maxFee * Number(gas) + Math.round(l1_fee_est * 1.25), // Optimism guarantees that there should not be one-time jumps of l1 component of gas estimation more than 25%
    estimatedFee: expectedFee * Number(gas) + l1_fee_est,
  };
```